src/medCore/medGlobal.cpp

```diff
@@ -0,0 +1,8 @@
+#include <medGlobal.h>
+#include <QCryptographicHash>
+
+QString anonymise(const QString name) {
+    QByteArray r = QCryptographicHash::hash(name.toUtf8(),QCryptographicHash::Md4).toHex().toUpper();
```

Comment:
Well, normally the probability of having a clash of keys is neglectible. The only thing is to ensure that if there is a name clash things do not crash. The simplest thing, we add a suffix (-1,-2,...) to the name. In this way, if we have the original data, we can still verify that it is conform and can avoid any name clashes. Looping on the hash, this will be difficult. The only thing is that we expose the order in which we added the patients to the database. If this is a problem, we generate a unique suffix randomly. But my bet is that we will never ever see a suffix being really useful.